volpe/annotate.py

```diff
@@ -12,10 +12,11 @@
 
 
 class AnnotateScope(Interpreter):
-    def __init__(self, tree: TypeTree, scope: Callable, args=None):
+    def __init__(self, tree: TypeTree, scope: Callable, args=None, stack_trace: List[TypeTree]=[]):
```

Comment:
I believe that reference type arguments with default are bad practice in python (pycharm complains).
This is because mutating the default argument will carry over to the next call where the default is used.

Reply:
Interesting, I didn't know that. I will use `None` as the default and will create a list `if stack_trace is None`.